### PR TITLE
Updating dapr to v1.7.0-rc.3

### PR DESCRIPTION
**Pull request by pravinpushkar**
# Description

Updating dapr to v1.7.0.-rc.3

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: NA

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -347,6 +347,8 @@ github.com/d2g/hardwareaddr v0.0.0-20190221164911-e7d9fbe030e4/go.mod h1:bMl4RjI
 github.com/danieljoos/wincred v1.1.0/go.mod h1:XYlo+eRTsVA9aHGp7NGjFkPla4m+DCL7hqDjlFjiygg=
 github.com/dapr/dapr v1.7.0-rc.1 h1:NEXZXuMHQZsd0I3Pi5bTJQdK1anDt0TRHLA1JzQnR78=
 github.com/dapr/dapr v1.7.0-rc.1/go.mod h1:YbacMSteh7nGRP777IJDuCn5sgJxL9cRVdiWEirBXbM=
+github.com/dapr/dapr v1.7.0-rc.3 h1:Qc1ezpsW5g2+AAi/X3zqJNxdMyifIqh8wpXRGbKaBPs=
```
Comment (surenderssm) [inline go.sum:350]: I don't see any changes in this file to remove reference of `v1.7.0-rc.1`? Am i missing something here?

Comment (shivamkm07) [reply go.sum:350]: It should have been removed. try running `go mod tidy -compat=1.17`. It will fix this.

**Review by mukundansundar — APPROVED** [review]

**Comment by mukundansundar** [comment]
@pravinpushkar Can you update the upgrade path tests also to use 1.7.0-rc.3?

**Review by mukundansundar — APPROVED** [review]
lgtm

**Pull request merged**